neuralpredictors/layers/cores/base.py

```diff
@@ -30,22 +22,42 @@ def __repr__(self):
 
 
 class Core2d(Core):
-    def initialize(self, cuda=False):
-        self.apply(self.init_conv)
-        self.put_to_cuda(cuda=cuda)
+    """
+    Base class for the core models, taking 2d inputs and computing nonlinear features.
+    """
 
-    def put_to_cuda(self, cuda):
-        if cuda:
-            self = self.cuda()
+    def initialize(self):
+        """
+        Initialization applied on the core.
+        """
+        self.apply(self.init_conv)
 
     @staticmethod
     def init_conv(m):
+        """
+        Initialize convolution layers with:
+            - weights: xavier_normal
+            - biases: 0
+
+        Args:
+            m (nn.Module): a pytorch nn module.
+        """
         if isinstance(m, nn.Conv2d):
             nn.init.xavier_normal_(m.weight.data)
             if m.bias is not None:
                 m.bias.data.fill_(0)
 
-    @staticmethod
-    def init_conv_hermite(m):
-        if isinstance(m, HermiteConv2D):
-            nn.init.normal_(m.coeffs.data, std=0.1)
+    def forward(self, x):
+        """
+        Forward function for pytorch nn module.
+
+        Args:
+            x (torch.tensor): input of shape (batch, channels, height, width)
+        """
+        raise NotImplementedError()
```

Comment:
Generally would be helpful for the `NotImplementedError` to give some useful message, at least pointing out what was expected